### PR TITLE
Restructure Getting Started documentation to be more compact and beginner friendly

### PR DESCRIPTION
**Pull request by malteneuss**
**Template Β: This PR does not modify `cabal` behaviour (documentation, tests, refactoring, etc.)**

Include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).

edit: I should wait with creating a PR until it is ready for review^^.

### CODE REVIEW

File: doc/getting-started.rst
Comment (ffaf1) [inline doc/getting-started.rst:4]: This is the only line I am a bit unsure about: it is true that `Installing Cabal` is simpler and more beginner friendly, but maybe the beginner is reassured by something like `Installing Cabal (and the Haskell toolchain)`.

File: doc/getting-started.rst
Comment (ffaf1) [inline doc/getting-started.rst:1]: Extreme nitpic, following line could be lenght-adjusted

File: doc/getting-started.rst
Comment (ffaf1) [inline doc/getting-started.rst:5]: lenght-adjusting here too.

**Review by ffaf1 — APPROVED** [review]

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst:4]: I would like the title to focus on Cabal (since it's the Cabal documentation). The Haskell toolchain is one method; We could mention the usual OS package managers (e.g. for Debian, Ubuntu, Mac, Windows, NixOS?) or how to install Cabal from source.

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst:1]: Done

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst]: Done

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:13]: The word "application" was deliberately chosen so that the newcomers know that this is the section they should read in order to learn how to create applications with cabal. The word "package", although technically ("more") correct, is intimidating: generally, newcomers don't look for creating packages in a broad sense (and as newcomers they don't know the precise technical sense in the case of cabal packages), because a package in a broad sense is a piece of reusable functionality, which is not what newcomers are after, most of the time.

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:24]: Ditto.

File: doc/getting-started.rst
Comment (geekosaur) [reply doc/getting-started.rst:197]: "The" seems correct there to me; plus I think "necessary" (insofar as they're needed for the script to work) in place of "important".

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:64]: "internal and external dependencies" is something not specified so far and doesn't really align with cabal's vocabulary I think, e g there are "internal libraries", yes, but I don't think the manual mentions "internal dependencies" anywhere (you could check though). I suggest KISS:
```suggestion
The ``build-depends`` section contains a list of package dependencies.
```

File: doc/getting-started.rst
Comment (ulysses4ever) [reply doc/getting-started.rst:197]: Thank you Brandon, I'm happy with your suggestion.

**Comment by ulysses4ever** [comment]
Thanks a lot for taking this on — this is a long-awaited action. But this is a tough one (partly the reason it hasn't been taken on for a while). I left several comments that I feel rather strong about, and that's only after a quick skimming. So, I expect much more word wrestling before it's merged. But rest assured this is a very welcome kind of contribution, and I'm hopeful it will get merged after all the struggle.

File: doc/getting-started.rst
Comment (andreabedini) [reply doc/getting-started.rst:13]: I feel strongly about making a distinction between a package and a project. A package is the unit of source code distribution. If you are a deploying applications, you are not distributing source code and you have a different set of problem from those who do.
I am on mobile and I dont remember the content of this section but "application" ~ project.

Comment (malteneuss) [reply doc/getting-started.rst:13]: Indeed. "application" seems to be target outcome of this mini-guide, which also fits the "myapp" package name.

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst]: Done

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst]: Done

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst]: Replaced "important" with "necessary"

**Comment by ulysses4ever** [comment]
@malteneuss I'll get back to this soon but currently I'm in a crunch.

**Comment by ulysses4ever** [comment]
I just wanna say this is still on my stack, but I'm afraid I'll have time to dig deeper only in the early October. I'm terribly sorry, but I do intend to get back to it unless, that is, someone else from the core team assists in driving this home.

**Comment by malteneuss** [comment]
No problem. It takes as long as it takes.

File: doc/getting-started.rst
Comment (huggablemonad) [inline doc/getting-started.rst:183]: I was kind of expecting the haskell-say example to be used here. It would feel more like a natural continuation of what has been written so far.

**Review by geekosaur — APPROVED** [review]

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst]: Good point. I made the examples the same.

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:45]: Maybe:
```suggestion
``myapp.cabal`` is a package description file (commonly referred as "a Cabal file"), which describes the package
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:46]:
```suggestion
its dependencies, and how it is built:
```
oxford commas FTW

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:77]:
```suggestion
To run the executable, switch into the project directory with ``cd myapp`` and  run 
```
`folder` is windows-speak.

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:63]:
```suggestion
The ``executable`` section shows that this package has a single component
of the executable type named ``myapp``.
```

Kinda obvious that executables can be run (from command line or otherwise)...

I'd prefer that the notion of `component` was introduced early on and used consistently across the text.

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:64]:
```suggestion
The ``build-depends`` section contains a list of the component's dependencies.
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:81]: I'm not sure I'm a big fan of this change. I'd rather (1) put the component's name back and (2) mention in the text that with one executable in the package, `cabal run` is smart enough to infer it, so the name can be omitted.

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:117]:
```suggestion
the executable section to include ``haskell-say``:
```

I think this will be easier on the reader. It's clear what we mean here anyway I think.

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:200]:
```suggestion
Use the sam ``cabal run`` command to execute this script:
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:206]:
```suggestion
On Unix-like systems, a script can be run directly with the execute permission.
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:233]: Let's ditch this detail? I think it's enough for a getting started session and all rest should be read in the respective manual section (handily referenced below).

**Comment by ulysses4ever** [comment]
At long last, I gave it another go. Let me know what you think...

**Comment by ulysses4ever** [comment]
And make sure to read the latest versions of suggestions, but the ones you see in email notifications: I edited them heavily.

**Review by ulysses4ever — CHANGES_REQUESTED** [review]
I put a block on it until the comments are resolved.

File: doc/getting-started.rst
Comment (ulysses4ever) [reply doc/getting-started.rst:46]: Actually, I think it'd be good to be a little more precise here and say that a package description contains package metadata and a sequence of sections, each defining a _package component_. Components can be of one of several types (executables, libraries, etc.). Every component describes, among other things, the location of source files and its dependencies. 

Could you integrate this description into the text?

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst:233]: Removed

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst:81]: Done

File: doc/getting-started.rst
Comment (malteneuss) [reply doc/getting-started.rst:46]: Done

**Comment by malteneuss** [comment]
Thanks for the suggestions. The guide now introduces the reader better into important packaging vocabulary. If everything looks good, you might want to squash the commits on merge

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:17]:
```suggestion
of the ``cabal`` commandline tool:
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:184]:
```suggestion
Cabal also supports running single-file Haskell scrips like 
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:67]:
```suggestion
Inside the ``executable`` section, the ``build-depends`` field lists the dependencies of this component.
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:80]:
```suggestion
To run the executable, switch into the application directory with ``cd myapp`` and  run 
```

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:120]:
```suggestion
In our ``myapp.cabal`` file we'll update the ``build-depends`` field of
```

**Comment by ulysses4ever** [comment]
@Mergify rebase

**Comment by ulysses4ever** [comment]
Remaining fixed one CI failure but introduced another one? @malteneuss can you look into this? https://github.com/haskell/cabal/actions/runs/6599697476/job/17929002660?pr=9212

**Review by ulysses4ever — APPROVED** [review]

**Comment by malteneuss** [comment]
@ulysses4ever Could you copy the error message? I can't see the reason the job is failing.

File: doc/getting-started.rst
Comment (ulysses4ever) [inline doc/getting-started.rst:45]:
```suggestion
The ``myapp.cabal`` file is a package description file, commonly referred to as a “Cabal file”:
```

**Comment by ulysses4ever** [comment]
Right. As per https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#github-pull-request-conventions you should feel free to apply the merge+squash label whenever you're ready or ask for permissions to apply labels or ask someone (e.g. me) to apply the label.

**Comment by malteneuss** [comment]
@ulysses4ever I hereby formally request rights to apply labels (e.g. to set "squash+merge me")

**Comment by ulysses4ever** [comment]
@malteneuss invited

**Pull request merged**
